site/components/DateSelector.tsx

```diff
@@ -1,81 +1,124 @@
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}${startOrEnd}DateDay`}
-                        type="text"
-                        defaultValue={inputs?.dayInput}
-                    />
-                </div>
-            </div>
-
-            <div className="govuk-date-input__item">
-                <div className="govuk-form-group">
-                    <label
-                        className="govuk-label govuk-date-input__label"
-                        htmlFor={type ? `${type}-${startOrEnd}-month-input` : `${startOrEnd}-month-input`}
-                    >
-                        Month
-                    </label>
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-2 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={`${type}-${startOrEnd}-month-input`}
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}DateMonth`}
-                        type="text"
-                        defaultValue={inputs?.monthInput}
-                    />
-                </div>
-            </div>
-            <div className="govuk-date-input__item">
-                <div className="govuk-form-group">
-                    <label
-                        className="govuk-label govuk-date-input__label"
-                        htmlFor={type ? `${type}-${startOrEnd}-year-input` : `${startOrEnd}-year-input`}
-                    >
-                        Year
-                    </label>
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-4 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={`${type}-${startOrEnd}-year-input`}
-                        name={`${type}${type ? startCase(startOrEnd) : startOrEnd}DateYear`}
-                        type="text"
-                        defaultValue={inputs?.yearInput}
-                    />
-                </div>
+const inputBox = (
+    startOrEnd: "start" | "end",
+    hasErrors: boolean,
+    inputRef: React.Ref<any> | undefined,
+    inputProps: InputBaseComponentProps | undefined,
+    InputProps: Partial<FilledInputProps> | Partial<OutlinedInputProps> | undefined,
+    disabled: boolean,
+    isPublishInfo: boolean
+) => (
+    <div className="govuk-date-input flex items-center" id={`${startOrEnd}-date`}>
+        <div className="govuk-date-input__item govuk-!-margin-right-0">
+            <div className="govuk-form-group">
+                <input
+                    className={`govuk-input govuk-date-input__input govuk-input--width-6 ${
+                        hasErrors ? "govuk-input--error" : ""
+                    } `}
+                    id={`${isPublishInfo && 'publish-'}${startOrEnd}-day-input`}
```

Comment:
This will render as "falsestart-day-input" when `isPublishInfo` is false, a ternary will work

Reply:
changed